src/index.ts

```diff
@@ -1,3 +1,43 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+interface MemoizeCacheItem {
+	// Memoize item status
+	status?: 'pending' | 'resolved' | 'rejected';
+
+	// Value if resolved
+	value?: any;
+
+	// Next request in waiting
+	next?: any;
+
+	// When this cache item was created
+	timestamp?: number;
+}
+
+type MemoizeCacheKey = string | number | symbol;
+
+type MemoizeUseCacheHandler = (param: MemoizeCacheItem) => boolean;
+
+interface MemoizeOptions {
+	// useCached: Will return the last resolved cached value
+	useCached?: true | false | MemoizeUseCacheHandler;
+
+	// staleInMs: How long before we should check for new updates
+	staleInMs?: number;
+
+	// getKey: Default key definition
+	getKey?: (...args: any) => MemoizeCacheKey;
```

Comment:
You can also weave the `ParameterTypes` from my earlier suggestion through to the options and use it there:

```suggestion
interface MemoizeOptions<ParameterTypes extends Array<unknown>> {
	// useCached: Will return the last resolved cached value
	useCached?: true | false | MemoizeUseCacheHandler;

	// staleInMs: How long before we should check for new updates
	staleInMs?: number;

	// getKey: Default key definition
	getKey?: (...args: ParameterTypes) => MemoizeCacheKey;
```